src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -160,11 +160,11 @@ public void delete(Tangle tangle) throws Exception {
 
     public List<Pair<Indexable, Persistable>> getMetadataSaveBatch() throws Exception {
         List<Pair<Indexable, Persistable>> hashesList = new ArrayList<>();
-        hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
-        hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
+        if(!getAddressHash().equals(Hash.NULL_HASH)) hashesList.add(new Pair<>(getAddressHash(), new Address(hash)));
+        if(getCurrentIndex() == 0) hashesList.add(new Pair<>(getBundleHash(), new Bundle(hash)));
```

Comment:
1. would make breaking changes to the API call `findTransactions` `bundles` - will only return tails, not all related hashes.
2. would require a DB migration (`--rescan`), not such an issue, just need to make sure it's clear.
--
we can postpone these changes (including not storing the `NULL_HASH`s) for another PR. it's not 100% related